### PR TITLE
1.9.3-p0 compatibility

### PR DESCRIPTION
**Pull request by thenrio**
Hello
I found ill behavior with mri-1.9.3-p0

  ~/src/ruby/wrong [master]$ ruby test/config_test.rb
    ruby 1.9.3p0 (2011-10-30 revision 33570) [x86_64-darwin10.8.0]
    /Users/thenrio/.rbenv/versions/1.9.3-p0/lib/ruby/1.9.1/rubygems/custom_require.rb:36:in `require': cannot load such file -- ParseTree (LoadError)
        from /Users/thenrio/.rbenv/versions/1.9.3-p0/lib/ruby/1.9.1/rubygems/custom_require.rb:36:in`require'
        from /Users/thenrio/src/ruby/wrong/lib/wrong/chunk.rb:7:in `require_optionally'
        from /Users/thenrio/src/ruby/wrong/lib/wrong/chunk.rb:13:in`<top (required)>'
        from /Users/thenrio/.rbenv/versions/1.9.3-p0/lib/ruby/1.9.1/rubygems/custom_require.rb:36:in `require'
        from /Users/thenrio/.rbenv/versions/1.9.3-p0/lib/ruby/1.9.1/rubygems/custom_require.rb:36:in`require'
        from /Users/thenrio/src/ruby/wrong/lib/wrong/assert.rb:5:in `<top (required)>'
        from /Users/thenrio/.rbenv/versions/1.9.3-p0/lib/ruby/1.9.1/rubygems/custom_require.rb:36:in`require'

WDYT?, Thierry

### CODE REVIEW

File: lib/wrong/chunk.rb [hunk 1]
```diff
@@ -6,7 +6,6 @@ def require_optionally(library)
   begin
     require library
   rescue LoadError => e
-    raise e unless e.message == "no such file to load -- #{library}"
```
Comment (alexch) [inline lib/wrong/chunk.rb:9]: what is the alternative LoadError message this is meant to trigger?

File: Gemfile [hunk 2]
```diff
@@ -5,6 +5,7 @@ gem "ruby2ruby"
 gem "sexp_processor"
 gem "predicated", '~> 0.2.1'
 gem "diff-lcs"
+gem "rake"
```
Comment (alexch) [inline Gemfile]: I'm reluctant to require rake as a non-development dependency. Even though _most_ people use wrong only for dev, and _most_ people use rake for dev, I'm sure there are others.

Comment (thenrio) [inline Gemfile]: good catch, this line is not required

(and it even does not change the behavior of `rake test`)

File: lib/wrong/chunk.rb (hunk 1 above)
Comment (thenrio) [inline lib/wrong/chunk.rb:9]: not sure to understand :)

## parse trees

`require "ParseTree"` fails on all rubies
`require "parse_tree"` fails on mri >= 1.9 and rubinius

(see source code `parse_tree.rb` :)

```
raise LoadError, "ParseTree doesn't work with ruby #{RUBY_VERSION}" if
  RUBY_VERSION >= "1.9"
raise LoadError, "ParseTree isn't needed with rubinius" if
  defined? RUBY_ENGINE and RUBY_ENGINE == "rbx"
```

## sourcify

```
begin
  require 'parse_tree'
  require 'parse_tree_extensions'
rescue LoadError
  # ParseTree is now optional for all supported rubies :)
end
```

so it looks like we can gleefully remove `require_optionally "ParseTree"`
:)

**Comment by thenrio** [comment]
So what about now ?
Curious and glad you reviewed this PR :)

**Comment by alexch** [comment]
Been too busy to dig into it. I tried installing 1.9.3rc1 via RVM and
got the dreaded readline library problem and haven't had time to try
again since.

On Wed, Nov 16, 2011 at 12:12 PM, Thierry Henrio
reply@reply.github.com
wrote:

> So what about now ?
> Curious and glad you reviewed this PR :)
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/sconover/wrong/pull/15#issuecomment-2767180

## 

Alex Chaffee - alex@cohuman.com - http://alexch.github.com
Stalk me: http://friendfeed.com/alexch | http://twitter.com/alexch |
http://alexch.tumblr.com

**Comment by thenrio** [comment]
Fun!

I took opportunity of 1.9.3-p0 to try rbenv... and had some other dreadful issues :)

(https://github.com/sstephenson/rbenv/issues/89 and gem pristine rake (or gem pristine rails...))

File: lib/wrong/chunk.rb [hunk 3]
```diff
@@ -6,11 +6,9 @@ def require_optionally(library)
...
   end
 end
 
-require_optionally "ParseTree"
```
Comment (mattscilipoti) [inline lib/wrong/chunk.rb:13]: Does this mean we can remove ParseTree from gemspec?

File: lib/wrong/chunk.rb (hunk 1 above)
Comment (mattscilipoti) [inline lib/wrong/chunk.rb:9]: While this line may no longer be required by ParseTree, is it stil needed by `require_optionally 'sourceify'`?

File: lib/wrong/chunk.rb (hunk 3 above)
Comment (thenrio) [inline lib/wrong/chunk.rb:13]: not yet there : there are 
- failing rspec tests (false negative)
- install issue (rake install)

removing dependency is a pain

**Comment by tjschuck** [comment]
Looks like this was fixed by 62ef049b9a73ad9f85e98325a0d975dd54da9409

**Comment by alexch** [comment]
@tjschuck true dat

**Pull request closed without merging**